be/src/runtime/lake_tablets_channel.cpp

```diff
@@ -530,6 +532,8 @@ void LakeTabletsChannel::add_chunk(Chunk* chunk, const PTabletWriterAddChunkRequ
             response->add_immutable_tablet_ids(tablet_id);
             response->add_immutable_partition_ids(writer->partition_id());
             immutable_tablet_ids.insert(tablet_id);
+
+            std::lock_guard l(_immutable_partition_ids_lock);
```

Comment:
if there is race condition here,  immutable_tablet_ids.count(tablet_id) == 0 doesn't hold the truth that _immutable_partition_ids doesn't have the tablet_id. will it cause problem that the `insert` in line 537 doesn't insert the tablet id into the set at all?

Reply:
Even if the result of immutable_tablet_ids.count(tablet_id) is incorrect, it will only cause the related immutable tablet not to be flushed to disk, and it will not lead to any correctness issues.